api/commands/run.ts

```diff
@@ -118,7 +120,19 @@ export class Runner {
 
       // Currently, we don't support caching for operations (and any dependents)
       await this.adapter.persistStateMetadata(
-        successfulActions.filter(action => action.type !== "operation")
+        successfulActions.filter(action => action && action.type !== "operation")
+      );
+      await this.adapter.deleteStateMetadata(
+        this.runResult.actions
+          .filter(
+            action =>
+              action.status !== dataform.ActionResult.ExecutionStatus.SUCCESSFUL &&
+              action.status !== dataform.ActionResult.ExecutionStatus.CACHE_SKIPPED
+          )
+          .map(action =>
+            this.graph.actions.find(executionAction => action.name === executionAction.name)
+          )
+          .filter(action => !!action)
```

Comment:
this line concerns me a bit. AFAIUI it should be impossible for `RunResult` to not contain an action entry for each action in `ExecutionGraph`